tests/phpunit/ConnectionTest.php

```diff
@@ -73,7 +76,36 @@ public function testRecursiveQueryWithoutTimeout(): void
 
         // long-running query
         $connection->executeQuery(
-            self::TIMEOUT_RECURSIVE_QUERY
+            self::TIMEOUT_RECURSIVE_QUERY,
         );
     }
+
+    public function testUserAgent(): void
+    {
+        $handlerStack = HandlerStack::create();
+
+        $requests = [];
+
+        $handlerStack->push(Middleware::mapRequest(function (Request $request) use (&$requests) {
+            $requests[] = $request;
+            return $request;
+        }));
```

Comment:
Tohle je presne `history` middleware https://docs.guzzlephp.org/en/stable/testing.html. Prislo by mi cistci ho pouzit, jestli tam neni nejaky skryty hacek.

Reply:
Není, určitě to zní víc clean. Upraveno.